common/flatpak-dir.c

```diff
@@ -4834,8 +4834,10 @@ flatpak_dir_pull (FlatpakDir                           *self,
           flatpak_remote_state_lookup_ref (state, ref, &rev, NULL, error);
           if (rev == NULL)
             {
-              g_set_error (error, FLATPAK_ERROR, FLATPAK_ERROR_INVALID_DATA, _("Couldn't find latest checksum for ref %s in remote %s"),
-                           ref, state->remote_name);
+              if (error != NULL && *error == NULL)
```

Comment:
Nitpick: This goes against the best practice of using `GError`, and while it should work OK, I think it might be cleaner to implement as the following (which assumes that `flatpak_remote_state_lookup_ref()` sets `error` if and only if it returns `FALSE`; please check that):
```
if (!flatpak_remote_state_lookup_ref (state, ref, &rev, NULL, error))
  goto out;
if (rev == NULL)
  {
    /* The lookup was successful, but the ref wasn’t found */
    g_set_error (error, … as below …);
    goto out;
  }
```

Reply:
I made essentially this change in several places in a separate commit, so that could use a quick review

Reply:
LGTM!